subt_ign/src/GameLogicPlugin.cc

```diff
@@ -767,7 +767,13 @@ void GameLogicPlugin::PostUpdate(
               auto filePath =
                 _ecm.Component<gazebo::components::SourceFilePath>(
                 model->Data());
-              this->dataPtr->robotSourceFilePaths.insert(filePath->Data());
+
+              // Store unique robot platform information.
+              for (const std::string &type : robotPlatformTypes)
+              {
+                if (filePath->Data().find(type) != std::string::npos)
```

Comment:
some of the robot configurations are not all uppercase characters, e.g. `Robotika_Freyja_Sensor_Config_1`, so in this case, it will not match with the string `FREYJA`  in `robotPlatformTypes`. Maybe convert `filePath->Data()` to uppercase before doing the check?

Reply:
Fixed: e58ad453ee9ec1efed14e72304cbcf505db45d52